internal/beatcmd/beat_test.go

```diff
@@ -192,7 +193,7 @@ func TestLibbeatMetrics(t *testing.T) {
 		snapshot = monitoring.CollectStructSnapshot(libbeatRegistry, monitoring.Full, false)
 		output := snapshot["output"].(map[string]any)
 		events := output["events"].(map[string]any)
-		return events["active"] == int64(0)
+		return events["active"] == int64(0) && events["batches"] == int64(totalRequests)
```

Comment:
nit: moving to [EventuallyWithT](https://github.com/stretchr/testify/blob/v1.11.1/assert/assertions.go#L2083) allows this to be split into 2 assert statements. If EventuallyWithT fails, it'll print more useful info.